### DISCUSSION TITLE
Washer&Dryer program

### DISCUSSION THREAD

**Discussion by pupe666** [ekutner/home-connect-hass#383] [Community Support]
TLDR; How can i use Home Connect API Service or how do I create a script that both washes and dries that ends in a secific time?

I have a washer/dryer that works perfect for the program that washes and sets the end time with just two presses on buttons in HA, instead of the buggy Home Connect App. Thanks for a super-great integration.

But... I have stumbled upon a problem. I want to start a program that both washes and dries. I only have one option for that dropdown list. That is a quick program that I do not want to use. According to https://api-docs.home-connect.com/programs-and-options/#washer-dryer the washer dryer has more than just the quick program. Could it be a limitation in the API for this specific appliance?

I have tried to use https://apiclient.home-connect.com/ to get information of the available programs for my appliance just to rule out that it is a integration-problem and is a API-problem. But unfortunately, my skills are not good enough. 
If I use the ID and secret for the application that is linked to HA I get an error after redirection. I tried to set up another application for test-purposes. But I still can’t authorize, just get a "client or redirection URI not authorized".  Anyone that knows how to use the Home Connect API Service? Or maybe some one knows a workaround to start a laundry-program and then a drying program that will be finished at a specific time?

And again, thanks for a great integration.

Kind regards 
Pontus
